### PR TITLE
🐛 Add go installs to Pinned-Dependencies score

### PR DESCRIPTION
**Pull request by gabibguti**
#### What kind of change does this PR introduce?

This is a bug fix. Just like pip and npm installs, go installs were identified but did not count for Pinned-Dependencies score. Now, they do.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

go installs are identified as pinned or unpinned and reported by Scorecard, but do not count for Pinned-Dependencies score.

#### What is the new behavior (if this is a feature change)?**

go installs now count in the aggregated score for Pinned-Dependencies.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

It's addressing one of the problems reported at https://github.com/ossf/scorecard/issues/2582

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
go installs count for Pinned-Dependencies score.
```

### CODE REVIEW

File: checks/evaluation/pinned_dependencies_test.go [hunk 1]
```diff
@@ -247,7 +247,23 @@ func Test_PinningDependencies(t *testing.T) {
...
+			name: "unpinned go install",
+			dependencies: []checker.Dependency{
+				{
+					Location: &checker.File{},
+					Type:     checker.DependencyUseTypeGoCommand,
+				},
+			},
+			expected: scut.TestReturn{
+				Error:         nil,
+				Score:         8,
```
Comment (spencerschrock) [inline checks/evaluation/pinned_dependencies_test.go:264]: Just an FYI: #3254 seems very relevant here.

File: checks/evaluation/pinned_dependencies_test.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func Test_PinningDependencies(t *testing.T) {
 				Error:         nil,
 				Score:         checker.MaxResultScore,
 				NumberOfWarn:  0,
-				NumberOfInfo:  7,
+				NumberOfInfo:  8,
```
Comment (spencerschrock) [inline checks/evaluation/pinned_dependencies_test.go:114]: similar fyi: this is also something that #3254 should probably fix.

**Review by spencerschrock — APPROVED** [review]
conversations are good to resolve, mainly leaving them as FYI comments. I think we should prioritize #3254 before any more forgotten ecosystems.

File: checks/evaluation/pinned_dependencies_test.go (hunk 1 above)
Comment (gabibguti) [reply checks/evaluation/pinned_dependencies_test.go:264]: Ack, I will prioritize https://github.com/ossf/scorecard/issues/3254 next!

File: checks/evaluation/pinned_dependencies_test.go (hunk 2 above)
Comment (gabibguti) [reply checks/evaluation/pinned_dependencies_test.go:114]: Ack!

**Pull request merged**